validphys2/src/validphys/config.py

```diff
@@ -723,24 +722,17 @@ def produce_dataset_inputs_fitting_covmat(
         """
         from validphys import covmats
 
-        if use_t0_fitting:
-            if theory_covmat_flag and use_thcovmat_in_fitting:
-                return covmats.dataset_inputs_t0_total_covmat
-            else:
-                return covmats.dataset_inputs_t0_exp_covmat
+        if theory_covmat_flag and use_thcovmat_in_fitting:
+            return covmats.dataset_inputs_t0_total_covmat
         else:
-            if theory_covmat_flag and use_thcovmat_in_fitting:
-                return covmats.dataset_inputs_total_covmat
-            else:
-                return covmats.dataset_inputs_exp_covmat
+            return covmats.dataset_inputs_t0_exp_covmat
```

Comment:
I would remove the `else` given the other changes.

Btw, why are we doing the import inside the method? Circular imports?

Reply:
Probable for no good reason anymore: At some point I wanted to make this file import quickly, but that ships has probably sailed.

Reply:
What do you mean exactly with removing the else? I would say we want support for enabling and disabling the theorycovmat. Not clear to me why there are two flags.

cc @andreab1997

Reply:
Just a stylistic comment, I would do

```python
if this:
    return that
return thet
```

The two flags are there because the th covmat can be used in the fit or in the sampling.
Since seeing the difference was relevant at some point I think that should stay.

Reply:
> The two flags are there because the th covmat can be used in the fit or in the sampling.
> Since seeing the difference was relevant at some point I think that should stay.

but the two flags are in the fitting action...

Reply:
One to say that the th covmat is active and another two to decide whether it is active in sampling or fitting.

A bit like the `use_t0` but in this case there is no the solution of "fitting without t0 is wrong, so let's remove the wrong option".

I guess that's the two flags you are referring to ?

Reply:
Indeed you can decide to use the thcovmat in only one of the two places or in both and I believe that one of Stefano's student will actually try to have a look to the different fits you can get with these three options.